routers/api/v1/repo/issue.go

```diff
@@ -389,6 +435,23 @@ func ListIssues(ctx *context.APIContext) {
 	ctx.JSON(http.StatusOK, convert.ToAPIIssueList(issues))
 }
 
+func getUserIDForFilter(ctx *context.APIContext, queryName string) int64 {
+	userName := ctx.Query(queryName)
+	if len(userName) != 0 {
+		user, err := models.GetUserByName(userName)
+		if err != nil {
+			if models.IsErrUserNotExist(err) {
+				ctx.NotFound(err)
+				return 0
+			}
+			ctx.InternalServerError(err)
+			return 0
+		}
+		return user.ID
+	}
+	return 0
+}
+
```

Comment:
This block is quite hard to read. Too many if nesting levels.

I suggest

```suggestion
func getUserIDForFilter(ctx *context.APIContext, queryName string) int64 {
	userName := ctx.Query(queryName)
	if len(userName) == 0 {
		return 0
	}

	user, err := models.GetUserByName(userName)
	if models.IsErrUserNotExist(err) {
		ctx.NotFound(err)
		return 0
	}

	if err != nil {
		ctx.InternalServerError(err)
		return 0
	}

	return user.ID
}

```